### PR TITLE
Enable FindBugs runs on "verify" stages + XML report generation

### PR DESCRIPTION
**Pull request by oleg-nenashev**
- [x] Reconfigure pom.xml
- [x] Enable FindBugs on https://jenkins.ci.cloudbees.com/job/plugins/job/ssh-slaves-plugin/
- [x] Fix important issues

The code is not clean according to the local FindBugs run

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
@reviewbybees
Fixed several issues, got timeboxed on others

File: src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java [hunk 1]
```diff
@@ -334,6 +358,10 @@ private static String hashClasspathEntry(URL entry) throws IOException {
                 return Util.toHexString(digest.digest());
             } catch (NoSuchAlgorithmException x) {
                 throw new AssertionError(x);
+            } finally {
+                if (input != null) {
+                    input.close();;
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java]: Unnecessary, this is only a wrapper stream, the base stream is only being closed, FindBugs is being dumb, suppress it.

File: src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java [hunk 2]
```diff
@@ -403,10 +433,13 @@ public synchronized String using(@Nonnull String script, @Nonnull Language langu
     /**
      * Called when configuring a classpath entry.
      * Usage is similar to {@link #configuring(String, Language, ApprovalContext)}.
-     * @param url the location of the entry
+     * @param entry entry, which is being configured
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java]: BTW incorrect use of English commas.

File: src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java [hunk 3]
```diff
@@ -778,10 +833,28 @@ public JSON denyApprovedClasspathEntry(String hash) throws IOException {
...
+    private static Jenkins getJenkins() throws IOException {
+        final Jenkins jenkins = Jenkins.getInstance();
+        if (jenkins == null) {
+            throw new IOException("Jenkins instance is not ready");
+        }
+        return jenkins;
+    }
+    
+    @Nonnull
+    private static Jenkins getActiveJenkinsInstance() throws IllegalStateException {
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java]: Should have a `TODO` comment to remove once the baseline supports `Jenkins.getActiveInstance`.

File: src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java (hunk 1 above)
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java]: I cannot suppress a single call, but suppressing the entire method is a hazardous thing. I would keep it as is (with a comment). Do you agree?

File: src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java (hunk 3 above)
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java]: It throws IOException here, but probably it's another case to Jenkins core

File: src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java (hunk 1 above)
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java]: Uhm, entry.openStream may open a FileInputStream. I think that the FindBugs complain was valid

File: src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java [hunk 4]
```diff
@@ -334,6 +358,10 @@ private static String hashClasspathEntry(URL entry) throws IOException {
                 return Util.toHexString(digest.digest());
             } catch (NoSuchAlgorithmException x) {
                 throw new AssertionError(x);
+            } finally {
+                if (input != null) {
+                    input.close();
+                }
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java]: BTW the simpler idiom (all local variables effectively final) is

```java
MessageDigest digest;
try {
    digest = MessageDigest.getInstance("SHA-1");
} catch (NoSuchAlgorithmException x) {
    throw new AssertionError(x);
}
DigestInputStream input = new DigestInputStream(new BufferedInputStream(is), digest);
try {
    // …
} finally {
    input.close();
}
```

which can also be mechanically converted more easily to a `-source 7` `try`-with-resource block.

File: src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java (hunk 3 above)
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java]: Not sure I understood that comment. Anyway I still request a

```java
// TODO 1.590+ use Jenkins.getActiveInstance
```

**Comment by jglick** [comment]
:bee:

File: src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java (hunk 3 above)
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java]: Never mind, you just put the comment above the other method.

**Pull request merged**